manifests/rules/docker_ce.pp

```diff
@@ -0,0 +1,124 @@
+# @summary Default firewall configuration for Docker-CE
+#
+# The configuration distributed in this class represents the default firewall
+# configuration done by docker-ce when the iptables integration is enabled.
+#
+# This class is needed as the default docker-ce rules added to ip-filter conflict
+# with the inet-filter forward rules set by default in this module.
+#
+# When using this class 'docker::iptables: false' should be set.
+#
+# @param docker_interface
+#   Interface name used by docker.
+# @param docker_prefix
+#   The address space used by docker.
+# @param manage_docker_chains
+#   Flag to control whether the class should create the docker related chains.
+# @param manage_base_chains
+#   Flag to control whether the class should create the base common chains.
+class nftables::rules::docker_ce (
+  String[1]                     $docker_interface     = 'docker0',
+  Stdlib::IP::Address::V4::CIDR $docker_prefix        = '172.17.0.0/16',
+  Boolean                       $manage_docker_chains = true,
+  Boolean                       $manage_base_chains   = true,
+) {
+  #
+  # inet-filter
+  #
+  if $manage_docker_chains {
+    nftables::chain {
+      'DOCKER': ;
+      'DOCKER_ISOLATION_STAGE_1': ;
+      'DOCKER_ISOLATION_STAGE_2': ;
+      'DOCKER_USER': ;
+    }
+  }
+
+  nftables::rule {
+    'DOCKER_ISOLATION_STAGE_1-iifname':
+      order   => '01',
+      content => "iifname \"${docker_interface}\" oifname != \"${docker_interface}\" counter jump DOCKER_ISOLATION_STAGE_2";
```

Comment:
same question as in the other PR: are the escaped quotes required / can we use single quotes?

Reply:
Normal interface names: eth0, docker0 could go without double quotes. However, if for some reason some uncommon interface name is used double quotes are needed. This approach should be safer for those uncommon cases.
There is an example of this in `man nft` describing the string type.